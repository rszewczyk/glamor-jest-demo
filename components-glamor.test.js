import React from 'react'
import renderer from 'react-test-renderer'
import {
  SomeInStylesheet,
  NoneInStylesheet,
  AllInStyleSheet
} from './components-glamor'
import serializer from 'jest-glamor-react'

expect.addSnapshotSerializer(serializer)

describe('component where all selectors are in the stylesheets', () => {
  it('renders', () => {
    expect(renderer.create(<AllInStyleSheet />).toJSON()).toMatchSnapshot()
  })
})

describe('component where some selectors are in the stylesheet', () => {
  it('renders', () => {
    expect(renderer.create(<SomeInStylesheet />).toJSON()).toMatchSnapshot()
  })
})

describe('component where none of the selectors are in the stylesheet', () => {
  it('renders', () => {
    expect(renderer.create(<NoneInStylesheet />).toJSON()).toMatchSnapshot()
  })
})
