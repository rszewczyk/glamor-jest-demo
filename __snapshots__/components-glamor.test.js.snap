// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component where all selectors are in the stylesheets renders 1`] = `
.glamor-1,
[data-glamor-1] {
  color: red;
}

.glamor-0,
[data-glamor-0] {
  display: inline;
}

<p
  className="glamor-0 glamor-1"
/>
`;

exports[`component where none of the selectors are in the stylesheet renders 1`] = `
<p
  className="css-foo css-bar container"
/>
`;

exports[`component where some selectors are in the stylesheet renders 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline;
}

<p
  className="glamor-0 glamor-1 container"
/>
`;
